src/config.rs

```diff
@@ -74,12 +74,34 @@ pub enum ConfigLoadError {
     Invalid(#[from] serde_yaml::Error),
 }
 
+#[derive(Clone, Debug, Deserialize, Default)]
+#[cfg_attr(feature = "json-schema", derive(schemars::JsonSchema))]
+#[serde(deny_unknown_fields)]
+#[serde(untagged)]
+#[cfg_attr(feature = "json-schema", schemars(with = "ThemeConfigSchema"))]
+pub enum ThemeConfig {
+    #[default]
+    None,
+    /// Theme of the presentation
+    Some(String),
+    /// Automatic dark/light theme switch based on the terminal background luminance
+    Dynamic {
+        /// Dark theme of the presentation
+        dark: String,
+        /// Light theme of ther presentation
+        light: String,
+        /// Light/Dark detection timeout in ms
+        #[cfg_attr(feature = "json-schema", validate(range(min = 1)))]
+        timeout: Option<u64>,
+    },
+}
+
 #[derive(Clone, Debug, Deserialize)]
 #[cfg_attr(feature = "json-schema", derive(schemars::JsonSchema))]
 #[serde(deny_unknown_fields)]
 pub struct DefaultsConfig {
     /// The theme to use by default in every presentation unless overridden.
-    pub theme: Option<String>,
+    pub theme: ThemeConfig,
```

Comment:
This needs a `serde(default)` otherwise on a config file that has `defaults` but doesn't have `theme` it fails.